server/src/main/java/com/thoughtworks/go/server/controller/JobController.java

```diff
@@ -111,17 +111,18 @@ public ModelAndView jobDetail(@RequestParam("pipelineName") String pipelineName,
                                   @RequestParam("stageName") String stageName,
                                   @RequestParam("stageCounter") String stageCounter,
                                   @RequestParam("jobName") String jobName) throws Exception {
-        if (!StringUtils.isNumeric(pipelineCounter)) {
-            throw bomb(String.format("Expected numeric pipelineCounter, but received '%s' for [%s/%s/%s/%s/%s]", pipelineCounter, pipelineName, pipelineCounter, stageName,
+        if (!StringUtils.isNumeric(pipelineCounter) && !JobIdentifier.LATEST.equalsIgnoreCase(pipelineCounter)) {
```

Comment:
I am wondering if all of this can be simplified a bit, how bout
 
```
@RequestMapping(value = "/tab/build/recent", method = RequestMethod.GET)
    public ModelAndView jobDetail(@RequestParam("pipelineName") String pipelineName,
                                  @RequestParam("pipelineCounter") String pipelineCounter,
                                  @RequestParam("stageName") String stageName,
                                  @RequestParam("stageCounter") String stageCounter,
                                  @RequestParam("jobName") String jobName) throws Exception {

        Optional<Integer> pipelineCounterValue = pipelineService.resolvePipelineCounter(pipelineName, pipelineCounter);

        if (!pipelineCounterValue.isPresent()) {
            throw bomb(String.format("Expected numeric pipelineCounter or latest keyword, but received '%s' for [%s/%s/%s/%s/%s]", pipelineCounter, pipelineName, pipelineCounter, stageName,
                    stageCounter, jobName));
        }

        if (!isValidCounter(stageCounter)) {
            throw bomb(String.format("Expected numeric stageCounter or latest keyword, but received '%s' for [%s/%s/%s/%s/%s]", stageCounter, pipelineName, pipelineCounter, stageName,
                    stageCounter, jobName));
        }

        Pipeline pipeline = pipelineService.findPipelineByNameAndCounter(pipelineName, pipelineCounterValue.get());

        if (pipeline == null) {
            throw bomb(String.format("Job %s/%s/%s/%s/%s not found", pipelineName, pipelineCounter, stageName,
                    stageCounter, jobName));
        }

        StageIdentifier stageIdentifier = restfulService.translateStageCounter(pipeline.getIdentifier(), stageName, stageCounter);

        JobInstance instance = jobInstanceDao.mostRecentJobWithTransitions(new JobIdentifier(stageIdentifier, jobName));
        return getModelAndView(instance);
    }

    private boolean isValidCounter(String counter) {
        return isNumeric(counter) || JobIdentifier.LATEST.equalsIgnoreCase(counter);
    }

```

If you are ok with these changes you will have to clean up rest of your changes.